### PR TITLE
Rest HL client: Add watcher stats API

### PR DESCRIPTION
**Pull request by imotov**
Relates #29827

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/WatcherRequestConverters.java [hunk 1]
```diff
@@ -115,4 +117,29 @@ static Request activateWatch(ActivateWatchRequest activateWatchRequest) {
...
+    static Request watcherStats(WatcherStatsRequest watcherStatsRequest) {
+        RequestConverters.EndpointBuilder builder =
+        new RequestConverters.EndpointBuilder()
+            .addPathPartAsIs("_xpack")
+            .addPathPartAsIs("watcher")
+            .addPathPartAsIs("stats");
+        String endpoint = builder.build();
+        Request request = new Request(HttpGet.METHOD_NAME, endpoint);
+        RequestConverters.Params parameters = new RequestConverters.Params(request);
+        StringBuilder  metric = new StringBuilder();
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/WatcherRequestConverters.java]: two spaces between `StringBuilder` and `metric`

File: client/rest-high-level/src/main/java/org/elasticsearch/client/WatcherRequestConverters.java [hunk 2]
```diff
@@ -115,4 +117,29 @@ static Request activateWatch(ActivateWatchRequest activateWatchRequest) {
         Request request = new Request(HttpPut.METHOD_NAME, endpoint);
         return request;
     }
+
+    static Request watcherStats(WatcherStatsRequest watcherStatsRequest) {
+        RequestConverters.EndpointBuilder builder =
+        new RequestConverters.EndpointBuilder()
+            .addPathPartAsIs("_xpack")
+            .addPathPartAsIs("watcher")
+            .addPathPartAsIs("stats");
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/WatcherRequestConverters.java]: addPathPartAsIs now takes a variadic

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/ExecutionPhase.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/ExecutionPhase.java]: extra emtpy line

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/WatcherStatsRequest.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+package org.elasticsearch.client.watcher;
+
+import org.elasticsearch.client.Validatable;
+
+/**
+ * A request to explicitly acknowledge a watch.
+ */
+public class WatcherStatsRequest implements Validatable {
+
+    private boolean includeCurrentWatches;
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/WatcherStatsRequest.java]: if there is no reason to make these mutable, lets make these final and have 2 constructors, a default one where we set them both to (im going to guess false, but thats up to you), and a 2 arg constructor where we ensure both are set.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/WatcherStatsResponse.java [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+
+    /**
+     * @return the status of the requested watch. If an action was
+     * successfully acknowledged, this will be reflected in its status.
+     */
+    public WatcherMetaData getWatcherMetaData() {
+        return watcherMetaData;
+    }
+
+
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/WatcherStatsResponse.java]: empty lines

File: client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/WatcherStatsResponse.java [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+        WatcherStatsResponse that = (WatcherStatsResponse) o;
+        return Objects.equals(nodes, that.nodes) &&
+            Objects.equals(header, that.header) &&
+            Objects.equals(clusterName, that.clusterName) &&
+            Objects.equals(watcherMetaData, that.watcherMetaData);
+    }
+
+    @Override
+    public int hashCode() {
+
```
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/watcher/WatcherStatsResponse.java]: empty line

File: client/rest-high-level/src/test/java/org/elasticsearch/client/watcher/WatcherStatsResponseTests.java [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+            this::toXContent,
+            WatcherStatsResponse::fromXContent)
+            .supportsUnknownFields(true)
+            .randomFieldsExcludeFilter(field -> field.endsWith("stats"))
+            .test();
+    }
+
+    private void toXContent(WatcherStatsResponse response, XContentBuilder builder) throws IOException {
+        builder.startObject();
+        NodesResponseHeaderTestUtils.toXContent(response.getHeader(), response.getClusterName(), builder);
```
Comment (hub-cap) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/watcher/WatcherStatsResponseTests.java]: This is only used here, do we need a separate class for it?

**Review by hub-cap — CHANGES_REQUESTED** [review]
mostly nits, good work

File: client/rest-high-level/src/test/java/org/elasticsearch/client/watcher/WatcherStatsResponseTests.java (hunk 7 above)
Comment (imotov) [reply client/rest-high-level/src/test/java/org/elasticsearch/client/watcher/WatcherStatsResponseTests.java]: I expect tests for other NodesAction-based rest actions will be using this as well in the future. We could have used it [here](https://github.com/elastic/elasticsearch/blob/a086c665a35af32e60e6d23ab6ca75fcf085b46f/client/rest-high-level/src/test/java/org/elasticsearch/client/security/ClearRolesCacheResponseTests.java#L41) for example if we had it handy.

**Review by hub-cap — APPROVED** [review]
:shipit:

**Pull request merged**